tk_toolchain/cmd_line_tools/tk_run_app/__init__.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
```

Comment:
So I'm using docopt here. docopt is a really sweet command line parser. Basically, you write your help just as you want it to show up using the unix convention for documenting command line arguments. Stuff like `[]` to indicate optional arguments, having `=<someting>` to specify that an option requires a value and so on.

It makes a great docstring because it explains how to exactly use the tool and then the `docopt` method can take that `__doc__` and pass it as a parameter to the parser.

Reply:
That's pretty sweet -- I haven't used it before, only `argparse`